OMCompiler/SimulationRuntime/c/simulation/solver/gbode_internal_nls.c

```
@@ -1156,57 +1425,84 @@ static NLS_SOLVER_STATUS gbInternalSolveNls_T_Transform(DATA *data,
 {
-  DATA_GBODE *gbData = (DATA_GBODE *)userData->solverData;
-  BUTCHER_TABLEAU *tabl = gbData->tableau;
+  BUTCHER_TABLEAU *tabl = (isFast ? ((DATA_GBODEF *) userData->solverData)->tableau
+                                  : ((DATA_GBODE *) userData->solverData)->tableau);
   T_TRANSFORM *trfm = tabl->t_transform;
   JACOBIAN* jacobian_ODE = &(userData->data->simulationInfo->analyticJacobians[userData->data->callback->INDEX_JAC_A]);
 
   GB_INTERNAL_NLS_DATA *nls = (GB_INTERNAL_NLS_DATA *) malloc(sizeof(GB_INTERNAL_NLS_DATA));
+
+  // multirate stuff
+  nls->multirate = isFast;
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)